azurerm/resource_arm_virtual_machine_managed_disks_test.go

```diff
@@ -256,6 +258,96 @@ func TestAccAzureRMVirtualMachine_changeStorageDataDiskCreationOption(t *testing
 	})
 }
 
+func TestAccAzureRMVirtualMachine_linuxNoConfig(t *testing.T) {
+	ri := acctest.RandInt()
+	config := testAccAzureRMVirtualMachine_linuxNoConfig(ri, testLocation())
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMVirtualMachineDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      config,
+				ExpectError: regexp.MustCompile("Error: either a `os_profile_linux_config` or a `os_profile_windows_config` must be specified."),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMVirtualMachine_windowsNoConfig(t *testing.T) {
+	ri := acctest.RandInt()
+	config := testAccAzureRMVirtualMachine_windowsNoConfig(ri, testLocation())
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMVirtualMachineDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      config,
+				ExpectError: regexp.MustCompile("Error: either a `os_profile_linux_config` or a `os_profile_windows_config` must be specified."),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMVirtualMachine_multipleNICs(t *testing.T) {
+	resourceName := "azurerm_virtual_machine.test"
+	ri := acctest.RandInt()
+	rs := acctest.RandString(5)
+	subscriptionId := os.Getenv("ARM_SUBSCRIPTION_ID")
+	prefix := fmt.Sprintf("/subscriptions/%s/resourceGroups/acctestRG-%d/providers/Microsoft.Network/networkInterfaces", subscriptionId, ri)
+	firstNicName := fmt.Sprintf("%s/acctni1-%d", prefix, ri)
+	secondNicName := fmt.Sprintf("%s/acctni2-%d", prefix, ri)
+
+	config := testAccAzureRMVirtualMachine_multipleNICs(ri, rs, testLocation())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMVirtualMachineDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "network_interface_ids.0", firstNicName),
+					resource.TestCheckResourceAttr(resourceName, "network_interface_ids.1", secondNicName),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMVirtualMachine_reorderedDisks(t *testing.T) {
+	var beforeVm compute.VirtualMachine
+	//var beforeVm, afterVm compute.VirtualMachine
```

Comment:
Is this comment needed?

Reply:
don't realise I'd pushed this last comma _sighs_ I'll remove it :)